calico/networking/configuring/qos-controls.mdx

```diff
@@ -0,0 +1,91 @@
+
+For example:
+
+```yaml noValidation
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-deployment
+  labels:
+    app: my-app
+spec:
+  replicas: 3
+  template:
+    metadata:
+      annotations:
+        qos.projectcalico.org/egressBandwidth: "100M"
+        qos.projectcalico.org/egressBurst: "200M"
+    spec:
+        (...)
+```
+
+## How to
+
+- [Configure bandwidth limits](#configure-bandwidth-limits)
+- [Configure packet rate limits](#configure-packet-rate-limits)
+- [Configure established connection limits](#configure-established-connection-limits)
+- [See all available annotations](#all-available-annotations)
+
+### Configure bandwidth limits
+
+To configure bandwidth limits,  use the `qos.projectcalico.org/ingressBandwidth` and `qos.projectcalico.org/ingressBurst` annotations for ingress, or `qos.projectcalico.org/egressBandwidth` and `qos.projectcalico.org/egressBurst` for egress, to configure bandwidth limits and burst sizes respectively. Bandwidth values are in bps (bits per second) and constrained from a minimum of `1kbps` to a maximum of `1Pbps` and burst values are in bits and constrained from a minimum of the value assigned to bandwidth in bits to a maximum of `4Gib` (2<sup>32</sup> bits). Burst configuration is optional, if burst is not explicitly configured with an annotation, a default value of `4Gib` is used.
+
+If the [Kubernetes bandwidth plugin](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/) annotations `kubernetes.io/ingress-bandwidth` or `kubernetes.io/egress-bandwidth` are configured and the `qos.projectcalico.org` annotations are not, they are honored and $[prodname] will configure bandwidth limits based on them . The $[prodname] specific annotations take precedence, so if both are configured, the Kubernetes annotation will be ignored. This makes upgrading from a $[prodname] version that uses the bandwidth plugin to a version with native bandwith limiting seamless, as the behavior will remain consistent across versions (with $[prodname] replacing the functionality previously provided by the Kubernetes bandwidth plugin).
+
+### Configure packet rate limits
+
+To configure packet rate limits,  use the `qos.projectcalico.org/ingressPacketRate` annotation for ingress, or `qos.projectcalico.org/egressPacketRate` for egress. Packet rate values are in packets per second and constrained from a minimum of `10` to a maximum of `1T` (10<sup>12</sup>) packets per second.
+
+### Configure established connection limits
+
+To configure limits for the number of established connections,  use the `qos.projectcalico.org/ingressMaxConnections` annotation for ingress, or `qos.projectcalico.org/egressMaxConnections` for egress. The values are in absolute numbers of simultaneous established connections and constrained from a minimum of `1` to a maximum of `100G` (100 * 10<sup>9</sup>) connections.
+
+### All available annotations
```

Comment:
feels like a duplicate of what you are doing in the 3 sections above? What if each of the sections has a small paragraph explaining in a sentence or two what the knobs are for followed by list of knobs that belong to that section (subset of the table) ?

Reply:
Good point, I can see that these sections are getting too long, will work on that